src/app/modules/main/node_section/view.nim

```diff
@@ -0,0 +1,188 @@
+    write = setCallResult
+    notify = callResultChanged
+
+  proc onSend*(self: View, inputJSON: string) {.slot.} =
+    self.setCallResult(self.delegate.sendRPCMessageRaw(inputJSON))
+    echo "Done!: ", self.callResult
+
+  proc onMessage*(self: View, message: string) {.slot.} =
+    self.setCallResult(message)
+    echo "Received message: ", message
+
+  proc receivedMessage*(self: View, lastMessage: string) {.signal.}
+
+  proc setLastMessage*(self: View, lastMessage: string) {.slot.} =
+    self.lastMessage = lastMessage
+    self.receivedMessage(lastMessage)
+
+  QtProperty[string] lastMessage:
+    read = lastMessage
+    write = setLastMessage
+    notify = receivedMessage
+
+  proc initialized(self: View) {.signal.}
+
+  proc getWakuBloomFilterMode*(self: View): bool {.slot.} =
+    result = self.wakuBloomFilterMode
+
+  proc getBloomLevel*(self: View): string {.slot.} =
+    if self.wakuBloomFilterMode and not self.fullNode:
+      return "normal"
+    if self.wakuBloomFilterMode and self.fullNode:
+      return "full"
+    return "light"
+
+  QtProperty[bool] wakuBloomFilterMode:
+    read = getWakuBloomFilterMode
+    notify = initialized
+
+  QtProperty[string] bloomLevel:
+    read = getBloomLevel
+    notify = initialized
+
+  proc setWakuBloomFilterMode*(self: View, bloomFilterMode: bool) {.slot.} =
+    self.delegate.setBloomFilterMode(bloomFilterMode)
+    quit(QuitSuccess) # quits the app TODO: change this to logout instead when supported
+
+  proc init*(self: View) {.slot.} =
+    self.wakuBloomFilterMode = self.delegate.getWakuBloomFilterMode()
+    self.fullNode = self.delegate.isFullNode()
+    self.wakuV2LightClient = self.delegate.isV2LightMode()
+    self.initialized()
+
+  proc wakuVersion*(self: View): int {.slot.} =
+    var fleetStr = self.delegate.getFleet()
+    let fleet = parseEnum[Fleet](fleetStr)
+    let isWakuV2 = if fleet == WakuV2Prod or fleet == WakuV2Test: true else: false
+    if isWakuV2: return 2
+    return 1
+
+  proc setBloomLevel*(self: View, level: string) {.slot.} =
+    var FullNode = false
+    var BloomFilterMode = false
+    case level:
+    of "light":
+      BloomFilterMode = false
+      FullNode = false
+    of "full":
+      BloomFilterMode = true
+      FullNode = true
+    else:
+      BloomFilterMode = true
+      FullNode = false
+
+    self.delegate.setBloomLevel(BloomFilterMode, FullNode)
+
+    quit(QuitSuccess) # quits the app TODO: change this to logout instead when supported
+
+  proc statsChanged*(self: View) {.signal.}
+
+  proc setStats*(self: View, stats: Stats) =
+    self.stats = stats
+    self.statsChanged()
+
+  proc fetchBitsSet*(self: View) =
+    self.delegate.fetchBitsSet()
+
+  proc getBloomBitsSet(self: View): int {.slot.} =
+    self.bloomBitsSet
+
+  proc bloomBitsSetChanged(self: View) {.signal.}
+
+  proc setBitsSet*(self: View, bitsSet: int) =
+    self.bloomBitsSet = bitsSet
+    self.bloomBitsSetChanged();
+
+  QtProperty[int] bloomBits:
```

Comment:
@borismelnik this `bloomBits` and corresponding methods are perfect, exactly as it should be.